src/Stormpath/Resource/Application.php

```diff
@@ -281,6 +286,74 @@ public function authenticate($username, $password, array $options = array())
     }
 
 
+    public function createIdSiteUrl(array $options = array())
+    {
+        if( ! isset( $options['callbackUri'] ) )
+            throw new InvalidCallbackUriException('Please provide a \'callbackUri\' in the $options array.');
+
+        $client = Client::getInstance();
+
+        $p = parse_url ( $this->href );
+        $base = $p['scheme'] . '://' . $p['host'];
+
+        $apiId = $client->dataStore->getRequestExecutor()->getApiKey()->getId();
+        $apiSecret = $client->dataStore->getRequestExecutor()->getApiKey()->getSecret();
+
+        $token = array(
+            'jti'       => UUID::v4(),
+            'iat'       => microtime(true),
+            'iss'       => $apiId,  //API ID
+            'sub'       => $this->href,
+            'state'     => isset($options['state']) ? $options['state'] : '',
+            'path'      => isset($options['path']) ? $options['path'] : '/',
+            'cb_uri'    => $options['callbackUri']
+        );
+
+        $jwt = JWT::encode($token, $apiSecret);
+
+        $redirectUrl = $base . "/sso";
+
+        if(isset($options['logout']))
+            $redirectUrl .= "/logout";
+
+        return $redirectUrl . "?jwtRequest=$jwt";
+
+    }
+
+    public function handleIdSiteCallback($responseUri)
+    {
+        $client = Client::getInstance();
```

Comment:
Same comment about not using the client from here and using the dataStore instead.